cel/cel_test.go

```diff
@@ -1825,6 +1825,107 @@ func TestDynamicDispatch(t *testing.T) {
 	}
 }
 
+func TestOptionalValues(t *testing.T) {
+	env, err := NewEnv(
+		OptionalTypes(true),
+		Variable("m", MapType(StringType, MapType(StringType, StringType))),
+		Variable("x", OptionalType(IntType)),
+		Variable("y", OptionalType(IntType)),
+		Variable("z", IntType),
+	)
+	if err != nil {
+		t.Fatalf("NewEnv() failed: %v", err)
+	}
+	tests := []struct {
+		expr string
+		in   map[string]interface{}
+		out  interface{}
+	}{
+		{
+			expr: `x.or(y).orValue(z)`,
+			in: map[string]interface{}{
+				"x": types.OptionalNone,
+				"y": types.OptionalNone,
+				"z": 42,
+			},
+			out: 42,
+		},
+		{
+			expr: `optional.ofNonZeroValue(z).or(optional.of(10)).value() == 42`,
+			in: map[string]interface{}{
+				"z": 42,
+			},
+			out: true,
+		},
+		{
+			expr: `(has(m.x) ? optional.of(m.x) : optional.none()).hasValue()`,
+			in: map[string]interface{}{
+				"m": map[string]map[string]string{},
+			},
+			out: false,
+		},
+		{
+			expr: `optional.ofNonZeroValue('').or(optional.of(m.c['dashed-index'])).orValue('default value')`,
```

Comment:
What's the intent of the test? that the middle optional gets applied? I think this might lead a reader to believe optional.of is doing some magic to check field presence.

Reply:
Done.